File: SpringBootFunctionalities/SpringSecurityPractice3/src/main/java/com/example/demo/entities/Trainer.java
package com.example.demo.entities;

import java.sql.Date;

import org.springframework.stereotype.Component;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
@Getter
@Setter
@NoArgsConstructor
@ToString
@Component
@Entity
@Table(name = "trainers")
public class Trainer {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int trainer_id;
	@Column
	private String fname;
	@Column
	private String lname;
	@Column
	private String email;
	@Column
	private String contactno;
	@Column
	private String specialization;
	@Column
	private int experience;
	@Column
	private Date registration_date;
	@JsonFormat(pattern = "yyyy-MM-dd")
	@Column
	private String address;
	@OneToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "user_id")
	private User user;
}

File: .NetFunctionalities/CaloryCoach/TrainerUseCases/Models/Trainer.cs
using System;
using System.Collections.Generic;

namespace TrainerUseCases.Models
{
    public partial class Trainer
    {
        public Trainer()
        {
            Customers = new HashSet<Customer>();
            TrainerRequests = new HashSet<TrainerRequest>();
        }

        public int TrainerId { get; set; }
        public string? Fname { get; set; }
        public string? Lname { get; set; }
        public string? Email { get; set; }
        public string? Contactno { get; set; }
        public string? Specialization { get; set; }
        public int? Experience { get; set; }
        public DateOnly? Dob { get; set; }
        public DateOnly? RegistrationDate { get; set; }
        public string? Gender { get; set; }
        public string? Address { get; set; }
        public int? UserId { get; set; }

        public virtual User? User { get; set; }
        public virtual ICollection<Customer> Customers { get; set; }
        public virtual ICollection<TrainerRequest> TrainerRequests { get; set; }
    }
}
